lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -194,31 +198,59 @@ defaultRecordTypeOptions = Aeson.defaultOptions
 getStakePoolMetadata
     :: Trace IO RegistryLog
     -- ^ Logging object - use 'transformTrace' to convert to 'Text'.
+    -> FilePath
+    -- ^ Directory to cache the metadata registry.
     -> String
     -- ^ URL of a zip archive to fetch and extract.
     -> [PoolOwner]
     -- ^ List of stake pools to get metadata for.
     -> IO (Either FetchError [Maybe StakePoolMetadata])
-getStakePoolMetadata tr url poolOwners = fmap join $ tryJust fileExceptionHandler $
-    withSystemTempFile "stake-pool-metadata.zip" $ \zipFileName zipFile -> do
-        let tr' = contramap (fmap (RegistryLog url zipFileName)) tr
-        logTrace tr' MsgDownloadStarted
-        fetchStakePoolMetaZip url zipFile >>= \case
-            Right size -> do
-                logTrace tr' $ MsgDownloadComplete size
-                hClose zipFile
-                Right <$> getMetadataFromZip tr' zipFileName poolOwners
-            Left e -> pure $ Left e
+getStakePoolMetadata tr cacheDir url poolOwners = do
+    let zipFileName = cacheDir </> "registry.zip"
+    let tr' = contramap (fmap (RegistryLog url zipFileName)) tr
+    createDirectoryIfMissing False cacheDir
+    fetchStakePoolMetaZipCached tr' url zipFileName >>= \case
+        Right _ -> Right <$> getMetadataFromZip tr' zipFileName poolOwners
+        Left e -> pure $ Left e
+
+cacheTTL :: NominalDiffTime
+cacheTTL = 3600
```

Comment:
**[suggestion]**

Add a comment about the duration. `NominalDiffTime` interprets `Int` as seconds for conversions through the `Num` instance, but that's not necessarily obvious :sweat_smile: